velox/functions/prestosql/aggregates/AverageAggregate.cpp

```diff
@@ -24,30 +24,39 @@ namespace facebook::velox::aggregate {
 
 namespace {
 
+template <typename TSumType>
 struct SumCount {
-  double sum{0};
+  TSumType sum{0};
   int64_t count{0};
 };
 
 // Partial aggregation produces a pair of sum and count.
-// Final aggregation takes a pair of sum and count and returns a real for real
+// Count is BIGINT() while sum  and the final aggregates type depends on
+// the input types:
+//       INPUT TYPE    |     SUM             |     AVG
+//   ------------------|---------------------|------------------
+//     REAL            |     DOUBLE          |    REAL
+//     SHORT_DECIMAL   |     LONG_DECIMAL    |    SHORT_DECIMAL
+//     LONG_DECIMAL    |     LONG_DECIMAL    |    LONG_DECIMAL
+//     ALL INTs        |     DOUBLE          |    DOUBLE
+//
 // input types and double for other input types.
 // T is the input type for partial aggregation. Not used for final aggregation.
-template <typename T>
+template <typename TInput, typename TAccumulator, typename ResultType>
```

Comment:
ResultType -> TResult

Reply:
We already have a TResult templating used at https://github.com/facebookincubator/velox/pull/2670/files#diff-013ebdfa6873d609e447826a2201cbf5e1a91c9f509cbe8f65452b5959cf2ad5R277

Reply:
Why do we need both then? Where is ResultType used?

Reply:
@mbasmanova Good catch! 
@karteekmurthys We don't need the `ResultType` here since.